src/github.com/getlantern/balancer/balancer.go

```diff
@@ -22,23 +22,42 @@ var (
 // Balancer balances connections established by one or more Dialers.
 type Balancer struct {
 	dialers []*dialer
+	trusted []*dialer
 }
 
 // New creates a new Balancer using the supplied Dialers.
 func New(dialers ...*Dialer) *Balancer {
 	dhs := make([]*dialer, 0, len(dialers))
+
 	for _, d := range dialers {
 		dl := &dialer{Dialer: d}
 		dl.start()
 		dhs = append(dhs, dl)
 	}
+
 	// Sort dialers by QOS (ascending) for later selection
 	sort.Sort(byQOSAscending(dhs))
+
 	return &Balancer{
 		dialers: dhs,
 	}
 }
 
+// trustedDialers returns the subset of b.dialers that are considered as
+// trusted.
+func (b *Balancer) trustedDialers() []*dialer {
+	if b.trusted == nil {
+		b.trusted = make([]*dialer, 0, len(b.dialers))
+		// Lazy initialization of trusted dialers.
```

Comment:
I was initially inclined to do that too, but I discovered e a couple of issues with that approach:

We need a set of all dialers (both trusted and untrusted) that we could use for encrypted traffic, and pick a subset of that set that only contains trusted dialers. Your snippet would create two mutually exclusive sets. That could be easily fixed, but we also need both sets to be sorted, we could do that with two different calls to `sort.Sort(byQOSAscending(trusted))` and `sort.Sort(byQOSAscending(all))` or we could create the `trusted` set after the set of all dialers is sorted, I decided over the later one just for clarity and moved the actual build of the list to another function to avoid having two `for` loops that looked too much alike in the same function (New), but now that I realize that a mutex needs to be created for that to work properly I think I'm going to avoid the mutex (and the complexity it carries) and just move this loop it inside New.